enterprise/server/util/oci/oci.go

```diff
@@ -113,7 +110,37 @@ func CredentialsFromProperties(props *platform.Properties) (Credentials, error)
 		}
 	}
 
-	return Credentials{}, nil
+	if !*useDefaultKeychain {
+		return Credentials{}, nil
+	}
+
+	// If no registries were found in the executor config, then fall back to the
+	// default keychain. This respects both docker configs (e.g.
+	// ~/.docker/config.json) and podman configs (e.g.
+	// ~/.config/containers/auth.json)
+
+	// TODO: parse the errors below and if they're 403/401 errors then return
+	// Unauthenticated/PermissionDenied
+	ctrRef, err := ctrname.ParseReference(ref.String())
+	if err != nil {
+		log.Debugf("Failed to parse image ref %q: %s", imageRef, err)
+		return Credentials{}, nil
+	}
+	authenticator, err := authn.DefaultKeychain.Resolve(ctrRef.Context())
+	if err != nil {
+		return Credentials{}, status.UnavailableErrorf("resolve default keychain: %s", err)
+	}
+	authConfig, err := authenticator.Authorization()
+	if err != nil {
+		return Credentials{}, status.UnavailableErrorf("authorize via default keychain: %s", err)
+	}
+	if authConfig == nil {
+		return Credentials{}, nil
+	}
+	return Credentials{
+		Username: authConfig.Username,
+		Password: authConfig.Password,
+	}, nil
```

Comment:
could we move this to a function like `resolveWithDefaultKeychain(ref)` ? Would simplify some of this logic i think

Reply:
Done